File: go/problems/day13_test.go
package problems

import "testing"

var day13 Solvable[int] = Day13{}

func TestDay13_TestSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day13_1")
	result := day13.Solve(input)

	var expected int = 405
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

func TestDay13_FullSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day13_2")
	result := day13.Solve(input)

	var expected int = 29213
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

File: ts/problems/day13.test.ts
import test, { describe } from "node:test";
import assert from "node:assert";

import solve from "./day13.js";
import { readProblemInput } from "../inputReader.js";

describe("Day 13", () => {
  test("test set should yield expected result", () => {
    const input = readProblemInput("day13_1");
    const result = solve(input);
    assert.equal(result, 405);
  });

  test("full set should yield expected result", () => {
    const input = readProblemInput("day13_2");
    const result = solve(input);
    assert.equal(result, 29213);
  });
});
